src/python_testing/TestSpecParsingSupport.py

```diff
@@ -106,6 +221,80 @@ def test_write_optional(self):
             asserts.assert_equal(xml_cluster.attributes[ATTRIBUTE_ID].write_optional,
                                  write_support == 'optional', "Unexpected write_optional value")
 
+    def test_derived_clusters(self):
+        clusters: dict[int, XmlCluster] = {}
+        derived_clusters: dict[str, XmlCluster] = {}
+        ids_by_name: dict[str, int] = {}
+        problems: list[ProblemNotice] = []
+        base_cluster_xml = ElementTree.fromstring(BASE_CLUSTER_XML_STR)
+        derived_cluster_xml = ElementTree.fromstring(DERIVED_CLUSTER_XML_STR)
+        expected_global_attrs = [GlobalAttributeIds.FEATURE_MAP_ID, GlobalAttributeIds.ATTRIBUTE_LIST_ID,
+                                 GlobalAttributeIds.ACCEPTED_COMMAND_LIST_ID, GlobalAttributeIds.GENERATED_COMMAND_LIST_ID, GlobalAttributeIds.CLUSTER_REVISION_ID]
+
+        add_cluster_data_from_xml(base_cluster_xml, clusters, derived_clusters, ids_by_name, problems)
+        add_cluster_data_from_xml(derived_cluster_xml, clusters, derived_clusters, ids_by_name, problems)
+
+        asserts.assert_equal(len(clusters), 1, "Unexpected number of clusters")
```

Comment:
This is going to be outside the scope of this PR, but I am curious. Do we have a mechanism to detected orphaned base clusters? Or is that not something we would even care about

Reply:
No, we don't. The support for derived clusters in the scraper isn't quite there yet (https://github.com/csa-data-model/projects/issues/390 - slated for fix in the 1.2.4 drop). Until that lands, we're still breaking fresh ground on base and derived.